### PR TITLE
Adding Changie 

### LINKED ISSUE #457: Bring MetricFlow in line with dbt-core CI/CD
**Describe the Feature**
dbt-core has done a lot of work to streamline their CI/CD. The biggest one that comes to mind is automating the changelog with changie. We should follow those practices with MetricFlow.

**Would you like to contribute?**
Yep!

### PR DESCRIPTION
**Pull request by callum-mcdata**
Resolves #457 

### This PR
- Adds `changie` to MetricFlow. `changie` is the automated changelog tool that the core team uses. Each PR will require that the user runs `changie new` that corresponds to a GH issue.
- Adds two new workflows for PRs, both used by changie.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (tlento) [inline .github/pull_request_template.md:19]: These feel like "first PR" items to me.

File: .github/pull_request_template.md
Comment (tlento) [inline .github/pull_request_template.md:20]: Will this apply to MetricFlow? We don't have a concept of running in development at the moment.

File: .github/pull_request_template.md
Comment (tlento) [inline .github/pull_request_template.md:22]: Oh I like this, but we are not set up for it yet. I'd prefer a link to the docs issue though.

File: .github/pull_request_template.md
Comment (tlento) [inline .github/pull_request_template.md:23]: The changie entry gets merged with the PR that contains it, right?

File: .github/pull_request_template.md
Comment (tlento) [inline .github/pull_request_template.md:9]: I noticed all of this in the dbt-core PRs. I'm not a fan of checklists in PR templates for a variety of reasons, but I'd like to learn more about the value they're providing in the dbt contributor space.

I would like our PR template to emphasize inclusion of why a change is being made, and why it is being made in this way. I think core would benefit from this as well.

File: CONTRIBUTING.md
Comment (tlento) [inline CONTRIBUTING.md:31]: We need these for now because not all of our tests work without them. Maybe we edit this line to something like: `OPTIONAL: if you want to run the full test suite you'll need to install some optional dbt dependencies. Any of the following commands should be sufficient:`

Alternatively, we can remove those tests, but I don't know what, if any, might be useful in the near future. @QMalcolm any thoughts here?

**Review by tlento — COMMENTED** [review]
Love the changie additions! Some questions:

1. Do you know what happens to our existing changelog? Is there something we need to do to preserve that content moving forward? I assume we can port it into changie without too much effort, but I've never done it before.
2. Some of the .github content seems internally focused to dbt labs. The ._README is an example, I don't think it needs to be in the MetricFlow repo
3. More broadly, I suggest splitting the `Other GH Workflow Changes` into a separate PR. The changie enabling seems really straightforward so I'd like to get that in place ASAP. Things like PR templates might need a little more discussion as we sort out the trade-offs between consistency in open source templates and customization for the different contexts in MF vs dbt-core, etc.

File: .github/_README.md
Comment (callum-mcdata) [reply .github/_README.md:19]: The actions repo is public and is used across our public repo's (I used actions from there in `dbt_metrics`) so I think this should be okay!

**Comment by callum-mcdata** [comment]
>Do you know what happens to our existing changelog? Is there something we need to do to preserve that content moving forward? I assume we can port it into changie without too much effort, but I've never done it before.

The existing changelog will remain as I've moved the contents into the `0.0.0.md` file that is used for backfilling information.

>Some of the .github content seems internally focused to dbt labs. The ._README is an example, I don't think it needs to be in the MetricFlow repo

I don't feel strongly one way or another on whether we include this! The only thought is that if we have external contributors it might be helpful context for them to understand Github Actions?

> More broadly, I suggest splitting the Other GH Workflow Changes into a separate PR. The changie enabling seems really straightforward so I'd like to get that in place ASAP. Things like PR templates might need a little more discussion as we sort out the trade-offs between consistency in open source templates and customization for the different contexts in MF vs dbt-core, etc.

Agreed I think this is completely reasonable.

File: .github/pull_request_template.md
Comment (callum-mcdata) [reply .github/pull_request_template.md:22]: I'm removing this section from the PR but @Jstein77 is already working on this!

File: .github/pull_request_template.md
Comment (callum-mcdata) [reply .github/pull_request_template.md:23]: Correct - it goes into the `unreleased` folder that then gets merged into the changelog when someone runs `changie batch` and `changie merge`.

File: CONTRIBUTING.md
Comment (callum-mcdata) [reply CONTRIBUTING.md:31]: Ah got it, I thought we had removed the needs for these tests! I'll add this code back in and then we can remove in another PR when the tests are removed.

File: .github/pull_request_template.md
Comment (callum-mcdata) [reply .github/pull_request_template.md]: Moving to another PR relevant to these changes!

**Comment by callum-mcdata** [comment]
@tlento I've reduced the scope of this PR to just adding changie and the information associated with it. I also moved the old CHANGELOG into `0.0.0.md` so that all of the information from before changie is retained.

**Review by tlento — APPROVED** [review]
This looks great, thank you!

I left some small suggestions inline. Apart from the CORE_TEAM thing they're all optional. The CORE_TEAM filter we should change out one way or another, but I'll leave that up to you if you want to do it before merging here or as part of a follow-up.

File: .changie.yaml
Comment (callum-mcdata) [reply .changie.yaml:136]: Definitely - I have to imagine there is probably some way we can automate that in the future.

File: .changie.yaml
Comment (callum-mcdata) [reply .changie.yaml:92]: Agreed - I'll accept your proposed commit below.

**Pull request merged**
